woocommerce/woocommerce
# Fix inconsistent order total on checkout vs manual order page

### All Submissions:


### Changes proposed in this Pull Request:

Fixes inconsistent order total on checkout vs manual order page when the shop is configured for tax-inclusive prices and a coupon is applied. Also refactors the `add_coupon_discount` and `calc_line_taxes` methods in the `WC_AJAX` class to make the code more unit testable.

Fix originally submitted by @Sidsector9 in https://github.com/woocommerce/woocommerce/pull/32010, work redone in a new pull request in order to easily apply ther required refactoring.

**Note for reviewers:** It's obscured by the refactor, but the main change in this pull request is executing `$order->calculate_taxes` and `$order->calculate_totals` _before_ `$order->apply_coupon`, instead of after, inside `add_coupon_discount`.

Closes #31971.

### How to test the changes in this Pull Request:

See https://github.com/woocommerce/woocommerce/issues/31971 and https://github.com/woocommerce/woocommerce/pull/32010

### Other information:




### FOR PR REVIEWER ONLY:
